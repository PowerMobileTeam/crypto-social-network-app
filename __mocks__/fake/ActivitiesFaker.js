
function load(count) {

  let activities = [...Array(count)].map((_, i) => {
    const code = 'activityguid' + i;
    return {
      attachment_guid: false,
      blurb: false,
      container_guid:code,
      custom_data:false,
      custom_type:false,
      description:"Congratulations! ",
      edited:"",
      guid:code,
      mature:false,
      ownerObj:{
        guid: "824853017709780997", 
        type: "user", 
        subtype: false, 
        time_created: "1522036284",
        getAvatarSource: () => {
          return {
            source:'http://thisisaurl'
          }
        }
      },
      message:"Message",
      title:'TITLE',
      owner_guid:"824853017709780997",
      parent_guid:"838106762591510528",
      perma_url:false,
      thumbnail_src:false,
      type:"activity",
    }
  });

  return {
    'activities': activities,
    'load-next': 'aaaaaa',
    'load-previous': 'aaaaaa'
  }
  
}

/**
 * comment service data faker
 * @param {integer} count
 */
export function activitiesServiceFaker() {
  return { load: load} 
}

