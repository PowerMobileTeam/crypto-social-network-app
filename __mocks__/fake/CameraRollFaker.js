
function buildPhoto(count) {

  const photos = [];

  return [...Array(count)].map((_, i) => {
    const code = Math.random().toString(16).substring(2).toUpperCase();
    return {
      'node': {
        'timestamp': Date.now(),
        'group_name': 'Camera Roll',
        'type': 'ALAssetTypePhoto',
        'location': {
          'speed': 2.0533344256922823,
          'latitude': 63.5314,
          'heading': 0,
          'longitude': -19.5112,
          'altitude': 0
        },
        'image': {
          'playableDuration': 0,
          'isStored': true,
          'filename': 'IMG_000'+i+'.JPG',
          'width': 3000,
          'height': 2002,
          'uri': 'assets-library://asset/asset.JPG?id=ED7AC36B-A150-4C38-BB8C-'+code+'&ext=JPG'
        }
      }
    }
  });
}

/**
 * getPhotos data faker
 * @param {integer} count
 */
export function getPhotosFaker(count = 3) {
  const photos = buildPhoto(count);
  return {
    'edges': photos,
    'page_info': {
      'has_next_page': false,
      'start_cursor': photos[0].node.image.uri,
      'end_cursor': photos[count-1].node.image.uri
    }
  };
}

