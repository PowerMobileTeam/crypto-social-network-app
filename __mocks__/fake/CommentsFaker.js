
function load(count) {

  let comments = [...Array(count)].map((_, i) => {
    let code = 'guid' +i;
    return {
      attachment_guid: false,
      blurb: false,
      container_guid:code,
      custom_data:false,
      custom_type:false,
      description:"Congratulations! ",
      edited:"",
      guid:code,
      mature:false,
      ownerObj:{
        guid: "824853017709780997", 
        type: "user", 
        subtype: false, 
        time_created: "1522036284"
      },
      owner_guid:"824853017709780997",
      parent_guid:"838106762591510528",
      perma_url:false,
      thumbnail_src:false,
      type:"comment",
    }
  });

  return {
    'comments': comments,
    'load-next': 'aaaaaa',
    'load-previous': 'aaaaaa'
  }
  
}

/**
 * comment service data faker
 * @param {integer} count
 */
export function commentsServiceFaker() {
  return { load: load} 
}

