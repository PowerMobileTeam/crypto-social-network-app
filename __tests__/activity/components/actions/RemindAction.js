import 'react-native';
import React from 'react';
import { Text, TouchableOpacity } from "react-native";
import { shallow } from 'enzyme';

import { activitiesServiceFaker } from '../../../../__mocks__/fake/ActivitiesFaker';

import renderer from 'react-test-renderer';
import RemindAction from '../../../../src/newsfeed/activity/actions/RemindAction';

import withPreventDoubleTap from '../../../../src/common/components/PreventDoubleTap';
import featuresService from '../../../../src/common/services/features.service';


describe('Thumb action component', () => {

  let screen;
  beforeEach(() => {

    const TouchableOpacityCustom = <TouchableOpacity onPress={this.onPress} />;

    const navigation = { navigate: jest.fn() };
    let activityResponse = activitiesServiceFaker().load(1);
    screen = shallow(
      <RemindAction entity={activityResponse.activities[0]} navigation={navigation} />
    );

    jest.runAllTimers();
  });

  it('renders correctly', async () => {
    screen.update();
    expect(screen).toMatchSnapshot();
  });


  it('should have a remind button', async () => {
    screen.update();
    expect(screen.find('PreventDoubleTap')).toHaveLength(1)
  });

  it('should navigate a thumb on press ', () => {
    let activityResponse = activitiesServiceFaker().load(1);

    const navigation = { 
      navigate: jest.fn() 
    };
    let entity = activityResponse.activities[0];
    entity.toggleVote = jest.fn();
    entity.votedUp = true;
    screen = shallow(
      <RemindAction entity={entity} navigation={navigation}/>
    );
    screen.update()
    let touchables = screen.find('PreventDoubleTap');
    touchables.at(0).props().onPress();
    jest.runAllTimers();

    expect(screen.instance().state.remindModalVisible).toBeTruthy();

  });

});