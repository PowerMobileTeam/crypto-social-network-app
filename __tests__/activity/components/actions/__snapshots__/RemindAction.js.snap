// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumb action component renders correctly 1`] = `
<PreventDoubleTap
  onPress={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      },
    ]
  }
>
  <Icon
    color="rgb(96, 125, 139)"
    name="repeat"
    raised={false}
    reverse={false}
    reverseColor="white"
    size={20}
    underlayColor="white"
  />
  <Counter
    size={15}
  />
  <Component
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ede3f2",
      }
    }
  >
    <Component
      animationType="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={false}
      visible={false}
    >
      <Component
        style={
          Object {
            "flex": 1,
            "paddingTop": 4,
          }
        }
      >
        <Component
          style={
            Object {
              "padding": 5,
            }
          }
        >
          <Icon
            allowFontScaling={false}
            color="gray"
            name="md-close"
            onPress={[Function]}
            size={30}
          />
        </Component>
        <RemindModal
          entity={
            Object {
              "attachment_guid": false,
              "blurb": false,
              "container_guid": "activityguid0",
              "custom_data": false,
              "custom_type": false,
              "description": "Congratulations! ",
              "edited": "",
              "guid": "activityguid0",
              "mature": false,
              "message": "Message",
              "ownerObj": Object {
                "getAvatarSource": [Function],
                "guid": "824853017709780997",
                "subtype": false,
                "time_created": "1522036284",
                "type": "user",
              },
              "owner_guid": "824853017709780997",
              "parent_guid": "838106762591510528",
              "perma_url": false,
              "thumbnail_src": false,
              "title": "TITLE",
              "type": "activity",
            }
          }
          onClose={[Function]}
        />
      </Component>
    </Component>
  </Component>
</PreventDoubleTap>
`;
